### ISSUE TITLE
Problems with Spectrum and presence updates from Conversations

### ISSUE THREAD

**Issue by dazzzl** [siacs/Conversations#2729]
#### General information

* **Version:** eg 1.23.2
* **Device:** Sony Xperia M  and Andorid Emulator
* **Android Version:** Lineageos 14.1 and Android 5.0
* **Server name:** self hosted
* **Server software:** prosody (trunk and prosody-modules from April tihs year)
* **Installed server modules:**  smacks, mam, csi, filter_chatstates, throttle_presence, smacks_offline, smacks_noerror
* **Conversations source:** PlayStore

#### Steps to reproduce

Due to lack of information inside conversations log and prosody log (maybe someone can tell me how to activate debug messages), I am logging the IP traffic on my server with wireshark. Due to enabled SSL encryption I see only encrypted packages. I only count the number of packages.

1. Bring conversations into foreground,
2. Switch conversations into backgound.
3. Look at the produced packages  on wireshark  

#### Expected result

Only few packages (<50) should be sent if the time period to the last foreground session is small  

#### Actual result

prosody-modules from April and conversations 1.23.2:

on Xperia M with Android 7.1: 140 - 250 packages -> Phone needs till 5 minutes to proceed these packages. In this time unable to receive new stanza ... Have to wait ~ 5min with display on to communicate with partner. Furthermore, increased traffic. 

on Android 5.0 with Android emulator: ~500 packages are generated.  

recent prosody-modules and conversations 1.23.2:

on Xperia M with Android 7.1: different behavior: After a new session (log off/ log on)  the number of packages is very small (<20). With every foreground/background cycle, it is increasing until >250 packages.

prosody-modules from April and conversations 1.18.5:

on Android 5.0 with Android emulator: less than 20 packages. This is what I would expect. 

So, something changes form Conversation 1.18.5 and the current version which trigger this behavior.

**Comment by iNPUTmice** [comment]
> Due to lack of information inside conversations log

`adb logcat -s conversations` should print enough information to get an idea of what is going on. Just post the output somewhere.

**Comment by dazzzl** [comment]
Not really ... :-(

12-18 17:52:34.263 24697 24697 D conversations: app switched into foreground
12-18 17:52:34.901 24697 13929 D conversations: xx@xx.net: making disco request for YS/ZEhtEnJLzdqfXAPGBnBAsKsI= to xx.net
12-18 17:53:30.961 24697 13929 D conversations: xx@xx.net: mismatch in caps for contact xx.net YS/ZEhtEnJLzdqfXAPGBnBAsKsI= vs QMLQY6n/ugPuPhk+JqN4mKUJnBM=
12-18 17:53:34.535 24697 24697 D conversations: app switched into background

**Comment by dazzzl** [comment]
I managed to get the debuug log from prosody... it is really huge. I can send this to you. But I do not want to post it here, because it is a high effort to anonymousize  it. A lot of this kind of messages are in the log:

Dec 18 18:22:15 c2sb0f7ec0      debug   #queue = 1062
Dec 18 18:22:15 c2sb0f7ec0      debug   Calling delayed_ack_function directly (still waiting for ack)

**Comment by dazzzl** [comment]
Ok, I investigated this behavior and this is related to a bug in Spectrum or Prosody [#256](https://github.com/SpectrumIM/spectrum2/issues/256). 

I activated the presence update if I use Conversation and I got a flood of presence updates from the gateways (~200 presence updates) Conversations needs minutes to process these updates. Even worse, If the screen went black the next flood of presence updates arrives. The result was that Conversations was not usable for more than 10 minutes and had a high impact on battery and mobile data amount. It was not usable, because the receiver queue was full and new messages and sending acknowledge arrived very late (sometimes more than 30 minutes later). 

The fault is clearly on the server, but maybe we could think about to prioritize messages and acknowledge stanzas over presence updates.
